ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+WKNavigationDelegate.swift

```diff
@@ -385,24 +384,28 @@ extension BrowserViewController: WKNavigationDelegate {
       BraveSearchManager.isValidURL(requestURL)
     {
 
-      // Add Brave Search headers if Rewards is enabled
-      if !isPrivateBrowsing && rewards.isEnabled
-        && navigationAction.request.allHTTPHeaderFields?["X-Brave-Ads-Enabled"] == nil
-      {
-        var modifiedRequest = URLRequest(url: requestURL)
-        modifiedRequest.setValue("1", forHTTPHeaderField: "X-Brave-Ads-Enabled")
-        tab?.loadRequest(modifiedRequest)
-        ContentBlockerManager.signpost.endInterval("decidePolicyFor", state, "Redirected to search")
-        return (.cancel, preferences)
-      }
-
       if let braveSearchResultAdManager = tab?.braveSearchResultAdManager,
         braveSearchResultAdManager.isSearchResultAdClickedURL(requestURL),
         navigationAction.navigationType == .linkActivated
       {
         braveSearchResultAdManager.maybeTriggerSearchResultAdClickedEvent(requestURL)
         tab?.braveSearchResultAdManager = nil
       } else {
+        // Add Brave Search headers if Rewards is enabled
+        if !isPrivateBrowsing && rewards.isEnabled
+          && navigationAction.request.allHTTPHeaderFields?["X-Brave-Ads-Enabled"] == nil
+        {
+          var modifiedRequest = URLRequest(url: requestURL)
+          modifiedRequest.setValue("1", forHTTPHeaderField: "X-Brave-Ads-Enabled")
+          tab?.loadRequest(modifiedRequest)
+          ContentBlockerManager.signpost.endInterval(
+            "decidePolicyFor",
+            state,
+            "Redirected to search"
+          )
+          return (.cancel, preferences)
+        }
+
```

Comment:
@aseren can you please provide logs before and after this change so that I can see the issue. Thanks

Reply:
Steps to reveal the problem:

1) Put log line `print("FOOBAR.requestURL: \(requestURL)")` before the following if statement:
```
    if navigationAction.targetFrame?.isMainFrame == true,
      BraveSearchManager.isValidURL(requestURL)
    {
```
2) Build and run the application.
3) Join Brave Rewards
4) Force show a search ad for brave search staging server.
5) Click a search ad.

Before the fix we get two log line when click a search ad, which is a bug.
```
FOOBAR.requestURL: https://search.brave.software/a/redirect?click_url=https%3A%2F%2Fbrave.com%2Fbrave-ads%2F%3Fno_key_param%3D&placement_id=7607875c-b340-4a22-a791-2268d13c7984&creative_instance_id=8483941c-c90a-4805-a6b3-454fff32410c&timestamp=1714599523&nonce=cf51a9fbd613d476f053b3faf3a82316&sig=cf42f68f955d7c2f9330af8f8addb6510a5fa03035672012f7bec6cc90c7b0fd
FOOBAR.requestURL: https://search.brave.software/a/redirect?click_url=https%3A%2F%2Fbrave.com%2Fbrave-ads%2F%3Fno_key_param%3D&placement_id=7607875c-b340-4a22-a791-2268d13c7984&creative_instance_id=8483941c-c90a-4805-a6b3-454fff32410c&timestamp=1714599523&nonce=cf51a9fbd613d476f053b3faf3a82316&sig=cf42f68f955d7c2f9330af8f8addb6510a5fa03035672012f7bec6cc90c7b0fd
```

After the fix we get only one log line when click a search ad, which is expected:
```
FOOBAR.requestURL: https://search.brave.software/a/redirect?click_url=https%3A%2F%2Fbrave.com%2Fbrave-ads%2F%3Fno_key_param%3D&placement_id=9b447f1d-4b68-4103-bfdc-ed884be6582e&creative_instance_id=8483941c-c90a-4805-a6b3-454fff32410c&timestamp=1714601193&nonce=142e6f7594c3501a63557d370501ce02&sig=27c019784a1f795cc4daf8d74a6e0aa54fc7ab803a908f85b550fbe216b70f20
```

The issue exists in stable and beta but not critical, because it is reproducible only for staging server when force showing ads.